src/monitor.js

```diff
@@ -33,14 +33,23 @@ class Monitor {
       return sentry;
     }).catch(err => {});
 
-    this._sentry.then(sentry => {
+    return this._sentry.then(sentry => {
       sentry.client.captureException(err, {
         tags: _.defaults({
           prefix: this._opts.project + (this._opts.prefix || '.root'),
           process: this._opts.process || 'unknown',
         }, tags),
         level,
       });
+
+      return new Promise((accept, reject) => {
+        sentry.client.once('logged', () => {
```

Comment:
Haha, yes. That's why I added the timeout to try to ensure that at the very least we crash.